x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/interceptor/SearchRequestInterceptor.java

```diff
@@ -74,4 +79,10 @@ private static boolean hasProfile(SearchRequest searchRequest) {
         return searchRequest.source() != null && searchRequest.source().profile();
     }
 
+    private static boolean hasZeroMinDocTermsAggregation(SearchRequest searchRequest) {
+        return searchRequest.source() != null
```

Comment:
I think we need to handle sub-aggregations here as well, do the same queue-based iteration we do elsewhere in this PR, right?

Reply:
Ah no ignore, `aggregations().hasZeroMinDocTermsAggregation()` takes care of that.